middleware/ratelimiter.go

```diff
@@ -140,7 +140,13 @@ func (t *tb) allow() bool {
 		alw = false
 	}
 
-	t.limit()
+	if alw {
+		t.messagesDelivered = t.messagesDelivered + 1
+		t.tokens = t.tokens - 1
+	} else {
+		t.limit()
+	}
```

Comment:
One effect of this new change is that if rate_limit = 100/sec and then you send requests that are equal to or less than 100, all of them will be let in immediately without delay.